### PR TITLE
fix source URLs for ant v1.10.5 - v1.10.7

### PR DESCRIPTION
**Pull request by Flamefire**
10.5 binary distribution is no longer available, 10.6/7 have https errors with wrong hostname as the url redirects to the new one

### CODE REVIEW

**Comment by Flamefire** [comment]
'https://apache.cu.be/ant/binaries' does not work as Python throws a ssl.CertificateError. Shouldn't this be handled by EB and other URLs tried?

**Comment by Flamefire** [comment]
Test report by @Flamefire
**FAILED**
Build succeeded for 2 out of 3 (3 easyconfigs in this PR)
taurusi6395.taurus.hrsk.tu-dresden.de - Linux RHEL 7.4, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/c385db2aa9fc893eb476c473c4b79061 for a full test report.

**Comment by Flamefire** [comment]
Some java packages seem to be missing when building the doc. However it is the same EC as other (older) ECs so I guess they work on someones system

**Comment by smoors** [comment]
Test report by @smoors
**FAILED**
Build succeeded for 3 out of 4 (3 easyconfigs in this PR)
node377.hydra.os - Linux centos linux 7.7.1908, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/3a12f6aafc8b7b99727a110ae61ca802 for a full test report.

**Comment by Flamefire** [comment]
@smoors Same issue. Can you try an older ant and see if that builds?

**Comment by Flamefire** [comment]
Bug seems to be known to others at least: https://stackoverflow.com/questions/56750209/how-to-successfully-build-ant-source-code. Building 10.1 does show errors but build succeeds. 10.5 shows more errors and build fails :/

**Comment by Flamefire** [comment]
And I traced the issue to a change in 10.2 which produces the following output on build not seen on 10.1:

```
Standard Doclet version 1.8.0_221
Building tree for all the packages and classes...
/home/h3/s3248973/apache-ant-1.10.2/src/main/org/apache/tools/ant/taskdefs/optional/net/FTP.java:1497: error: reference not found
     * @see org.apache.commons.net.ftp.FTPClientConfig
            ^
```

I then found that the issue is a removed `-Xdoclint:none` from javadoc command line which runs into https://bugs.openjdk.java.net/browse/JDK-8224266 and similar to https://github.com/hamcrest/JavaHamcrest/issues/75

**Comment by Flamefire** [comment]
Patch added, built successfully here :)

File: easybuild/easyconfigs/a/ant/ant-fix-doclint-error.patch [hunk 1]
```diff
@@ -0,0 +1,11 @@
+diff -aur a/build.xml b/build.xml
```
Comment (boegel) [inline easybuild/easyconfigs/a/ant/ant-fix-doclint-error.patch]: @Flamefire Please include a brief patch description, with references to relevant upstream bug issues or PRs (or a patch author)

Comment (Flamefire) [reply easybuild/easyconfigs/a/ant/ant-fix-doclint-error.patch]: "Author" is me, not sure if it is worth mentioning for 1 line (re)added ;)
Where/how would I add the description?

Comment (boegel) [reply easybuild/easyconfigs/a/ant/ant-fix-doclint-error.patch]: Just on top, anything above the 1st `diff` line is comments.

Comment (Flamefire) [reply easybuild/easyconfigs/a/ant/ant-fix-doclint-error.patch]: Ok, done

File: easybuild/easyconfigs/a/ant/ant-1.10.5-Java-1.8.eb [hunk 2]
```diff
@@ -1,24 +1,29 @@
...
+
+# specify dependency on Java/1.8 "wrapper", rather than a specific Java version
+dependencies = [('Java', '1.8', '', True)]
 
-dependencies = [('Java', '1.8')]
+builddependencies = [('JUnit', '4.12', versionsuffix)]
 
 sanity_check_paths = {
-    'files': ['bin/ant', 'lib/ant.jar'],
+    'files': ['bin/ant', 'lib/ant.jar', 'lib/ant.jar'],
```
Comment (boegel) [inline easybuild/easyconfigs/a/ant/ant-1.10.5-Java-1.8.eb]: @Flamefire Why check the same file twice?

File: easybuild/easyconfigs/a/ant/ant-1.10.5-Java-1.8.eb [hunk 3]
```diff
@@ -1,24 +1,29 @@
...
  Java applications."""
 
 toolchain = SYSTEM
 
-source_urls = ['http://apache.cu.be//ant/binaries']
-sources = ['apache-%(name)s-%(version)s-bin.tar.gz']
-checksums = ['cd52c6f77a0b1de7174addc94e5a512d86978ac9173a30cd11f4ab26f6d8deba']
+source_urls = ['https://archive.apache.org/dist/%(name)s/source/']
+sources = ['apache-%(name)s-%(version)s-src.tar.gz']
+patches = ['ant-fix-doclint-error.patch']
```
Comment (boegel) [inline easybuild/easyconfigs/a/ant/ant-1.10.5-Java-1.8.eb]: please rename the patch file to include the `ant` version, for example `ant-1.10.5_fix-doclint-error.patch`

File: easybuild/easyconfigs/a/ant/ant-1.10.5-Java-1.8.eb [hunk 4]
```diff
@@ -1,24 +1,29 @@
...
  build files as targets and extension points dependent upon each other. The main known usage of Ant is the build of
  Java applications."""
 
 toolchain = SYSTEM
 
-source_urls = ['http://apache.cu.be//ant/binaries']
-sources = ['apache-%(name)s-%(version)s-bin.tar.gz']
-checksums = ['cd52c6f77a0b1de7174addc94e5a512d86978ac9173a30cd11f4ab26f6d8deba']
+source_urls = ['https://archive.apache.org/dist/%(name)s/source/']
+sources = ['apache-%(name)s-%(version)s-src.tar.gz']
```
Comment (boegel) [inline easybuild/easyconfigs/a/ant/ant-1.10.5-Java-1.8.eb]: Why is this changed to `-src.tar.gz` while it's left to `-bin.tar.gz` for `ant` 1.10.6 & 1.10.7?

Comment (Flamefire) [reply easybuild/easyconfigs/a/ant/ant-1.10.5-Java-1.8.eb]: Because for 10.6+ there still is a -bin package

File: easybuild/easyconfigs/a/ant/ant-1.10.5-Java-1.8.eb (hunk 2 above)
Comment (Flamefire) [reply easybuild/easyconfigs/a/ant/ant-1.10.5-Java-1.8.eb]: C&P from old one :/

File: easybuild/easyconfigs/a/ant/ant-1.10.5-Java-1.8.eb (hunk 4 above)
Comment (boegel) [reply easybuild/easyconfigs/a/ant/ant-1.10.5-Java-1.8.eb]: OK, thanks for clarifying!

**Comment by Flamefire** [comment]
@boegel "source URLs" is wrong as binary URLs are fixed too

**Comment by boegel** [comment]
@Flamefire With "source URLs" I mean the value for the `source_urls` easyconfig parameter.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
node2509.golett.os - Linux centos linux 7.7.1908, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/0ceed7fdeabbc6baf6aec1f0ba27b1fa for a full test report.

**Comment by boegel** [comment]
Going in, thanks @Flamefire!
